src/components/LoginForm.js

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from 'react';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import { FaFacebookF, FaGoogle } from 'react-icons/fa';
+import styles from '../styles/entry.module.css';
+import axios from 'axios';
+import Snackbar from '@material-ui/core/Snackbar';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+import { connect } from 'react-redux';
+import { loginAction, closeMessage } from '../redux/actions/loginAction';
+import { LinearProgress } from '@material-ui/core';
+
+class LoginForm extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			email: '',
+			emailError: '',
+			emailErrorStatus: false,
+			password: '',
+			passwordError: '',
+			passwordErrorStatus: false,
+			open: false,
+			errorMessage: ''
+		};
+	}
+	validate = () => {
+		let isError = false;
+		const errors = {
+			emailError: '',
+			emailErrorStatus: false,
+			passwordError: '',
+			passwordErrorStatus: false
+		};
+
+		if (this.state.email.indexOf('@') === -1) {
+			isError = true;
+			errors.emailErrorStatus = true;
+			errors.emailError = 'Requires valid email';
+		}
+		if (this.state.password.length < 1) {
+			isError = true;
+			errors.passwordErrorStatus = true;
+			errors.passwordError = 'Please provide a password';
+		}
+		this.setState({
+			...this.state,
+			...errors
+		});
+
+		return isError;
+	};
+
+	onSubmit = async (e) => {
+		e.preventDefault();
+		const error = this.validate();
+		if (!error) {
+			const send = {
+				email: this.state.email,
+				password: this.state.password
+			};
+			await this.props.loginAction(send);
+		}
+	};
+
+	change = (e) => {
+		this.setState({
+			[e.target.name]: e.target.value
+		});
+	};
+	render() {
+		return (
+			<div>
+				<div className={styles.forms}>
+					<div style={{ display: this.props.loginSate.loading }}>
+						<LinearProgress />
```

Comment:
@blaise82  can you consider redirecting a user for a successful login attempt(Say redirect to update profile, even though the page is not there, please do include such route)? Please do include it in your Login Component .